src/SFML/Window/Android/JoystickImpl.hpp

```diff
@@ -94,13 +94,19 @@ class JoystickImpl
     /// \return Joystick state
     ///
     ////////////////////////////////////////////////////////////
-    [[nodiscard]] JoystickState update();
+    [[nodiscard]] JoystickState update() const;
+
+    [[nodiscard]] static std::optional<Joystick::Axis> androidAxisToSf(int axisCode);
+
+    [[nodiscard]] static int sfAxisToAndroid(Joystick::Axis axis);
 
 private:
     ////////////////////////////////////////////////////////////
     // Member data
     ////////////////////////////////////////////////////////////
     Joystick::Identification m_identification; ///< Joystick identification
+    JoystickCaps             m_capabilities;
+    int32_t                  m_currentDeviceIdx = -1; ///< Physical device ID
```

Comment:
Maybe instead of using `-1` as surrogate, we could use `std::optional` instead.

Reply:
changed